libretro-common/audio/audio_mixer.c

```diff
@@ -40,23 +40,23 @@
 #define STB_VORBIS_NO_STDIO
 #define STB_VORBIS_NO_CRT
 
-#include <stb/stb_vorbis.h>
+#include "stb_vorbis.h"
 #endif
 
 #ifdef HAVE_DR_FLAC
 #define DR_FLAC_IMPLEMENTATION
-#include <dr/dr_flac.h>
+#include "dr_flac.h"
```

Comment:
I don't agree with the changes here. It would just confuse developers where these headers are coming from, instead of it being clear that they come from system headers.

If you don't want these to conflict with system header files that is one thing, but I'd suggest them renaming our internal header files to something else so that it won't conflict. 

Either way, it strikes me as a fundamentally bad thing to do that will just further confuse developers instead of things becoming more clear.

Reply:
Between `find(1)` and `grep(1)` this is entirely a non-issue and modifying included builtin dependencies is a poor suggestion as it just increases the technical debt and makes it harder for future developers to update them cleanly.

The best solution is to remove the long standing hack of including the entire `deps/` directory, this cleanly fixes the issue and prevents it from occurring again in other places.

Look, I'm not opposed to spending time making RetroArch better, but its not possible when I'm constantly having my hands tied behind my back.

Reply:
I can live with a solution like this -

https://github.com/libretro/RetroArch/blob/master/gfx/drivers_shader/glslang.cpp#L20

just make sure that in the HAVE_BUILTIN path, that you use relative path includes - i.e. when HAVE_BUILTIN is defined, it would be best to go about it by not requiring any include dirs. I think this way you can have your cake and eat it - i.e. by not setting HAVE_BUILTIN, it would look inside system dirs instead.

Reply:
I don't really like that, but its beyond the point. I think you misunderstood the problem. The problem is that with `-Ideps/` it will always check the `deps/` directory before any system include paths like `/usr/include/`. So for example with `miniupnpc` it will always find `deps/miniupnpc/miniwget.h` before the `/usr/include/miniupnpc/miniwget.h`. This is a rather unfortunate problem if the user has set `---disable-builtinminiupnpc` and no matter how the builtin include path is written it will exist.

The important part is that `-Ideps/` has more potential for harm than convenience.

As for the point you raised, setting a relative path to `deps/` in `libretro-common` seems wrong and means that downstream projects that use `libretro-common` have to put their builtin `miniupnpc` directory in a specific location, but its not really important for this specific PR.

Reply:
I think this is all way too pedantic for my taste.

Let me just boil it down to what I don't want to see -

> #include "dr_flac.h"

I don't want to see this. Either it needs to be a relative path include that corresponds exactly to where the source file is relative to the header file, or we look for it in the system dir. I already explained how I want to see it. Again, work with defines like HAVE_BUILTIN and if you don't want the builtin version, then just don't define it. We will want most normal RetroArch versions to be using the builtin versions anyway, but we want to keep the option open for people to roll their own versions with system-provided dependencies instead.

I really don't understand what the big problem here is or what needs to be fixed here in your view since I really don't see the problem, but I'm always open for allowing the user the extra flexibility to configure RetroArch to his/her liking during the compilation process.

Reply:
This is exactly what I mean by tying my hands behind my back. Instead of just fixing issues I am given the hopeless and tedious task of educating you on how and why I am making changes. No offense, but you don't seem to be absorbing any of it and I'm not all that interested in fighting you over it. Frankly I rather spend the time just being productive, but you are in the way so I am forced to spend my efforts on other things...

Reply:
If you want to discuss things then come to Discord where the developers hang out, or IRC. This is not the proper venue for it.

To be very frank with you, I have been very understanding and accomodating with these increasingly minuteau Makefile refactors and tweaks. But if you are not able to convince me of the reason behind some of the more intrusive changes, I don't see why we have to incorporate it. If you struggle to make your point without resorting to insults and slights, I don't see any further need to correspond with you. I hope that you will reflect back on this increasingly unproductive behavior and take the necessary steps so that we can get more productive. Either that or we leave it like this, I'm not interested in a constant passive aggressive back and forth with you.

Again, I have no issues with you, and I'm hoping that this unnecessary bumping of heads won't remain the default mode of interaction from this point forward.